src/emm/services/modules_service.py

```diff
@@ -135,3 +136,51 @@ def sync_module(self, module_name: str, module_data: EvgModule) -> None:
 
         self.git_service.fetch(module_location)
         self.git_service.checkout(module_revision, directory=module_location)
+
+    def attempt_git_operation(
+        self,
+        operation: GitAction,
+        revision: str,
+        branch_name: Optional[str] = None,
+        directory: Optional[Path] = None,
+    ) -> Optional[str]:
+        """
+        Attempt to perform the specified git operation.
+
+        :param operation: Git operation to perform.
+        :param revision: Git revision to perform operation on.
+        :param branch_name: Name of branch for git checkout.
+        :param directory: Directory of git repository.
+        :return: Error message if an error was encountered.
+        """
+        try:
+            self.git_service.perform_git_action(operation, revision, branch_name, directory)
+        except ProcessExecutionError:
```

Comment:
I don't think we should just swallow this exception. I think we should at least log the exception so that users can debug why the failure occurred.